### DISCUSSION TITLE
Applying events to `IEventStream<T>` does not update the aggregate

### DISCUSSION THREAD

**Discussion by martinothamar** [JasperFx/marten#2591] [General]
Hi!

I have an aggregate/stream where I receive a batch of commands that I wan't to execute against an aggregate in a single transaction. I use the `IEventStream<T>` methods for this. Some commands may be dependent on the result of the previous commands in a single batch, which doesn't work since the aggregate is never updated. If the stream doesn't exist from before (i.e. the command batch element 0 is `CreateThing`) then `IEventStream<T>.Aggregate` is always null as weel.

I can work around this of course by manually applying events and keeping track of the aggregate outside `IEventStream<T>` (since `Aggregate` property is not settable), but I thought I'd raise an issue since it surprised me a bit

**Comment by jeremydmiller** [comment]
@martinothamar That's not a use case that Marten supports. The `IEventStream` is meant completely for short lived transactions and was never meant to be used outside of that context. Sorry, but Marten just doesn't have any direct support for what you're describing here.

I'm converting this to a discussion as I don't feel like this is a bug but a feature request for the future.
